diff --git a/src/app.js b/src/app.js
index fc64d70..e716374 100644
--- a/src/app.js
+++ b/src/app.js
@@ -706,6 +706,7 @@ async function resolveComponents(rt, options, state, root, report) {
       if (state?.inProgress?.versions[name] && !options.refreshPending) {
         versions[name] = state.inProgress.versions[name];
       } else if (options.command !== "upgrade" && state?.components?.[name]?.version) {
+        await resolveVersion(rt, name);
         versions[name] = state.components[name].version;
       } else if (name === "assertledger" && options.command === "setup") {
         let existing;
